sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -407,17 +415,257 @@ public Object convertDate(java.util.Date date, int sqlType, Type type) throws Ap
         if (date != null) {
             switch (type.getTag()) {
                 case TypeTags.STRING_TAG:
-                    return fromString(Utils.getString(date));
+                    return fromString(date.toString());
                 case TypeTags.OBJECT_TYPE_TAG:
                 case TypeTags.RECORD_TYPE_TAG:
-                    return Utils.createTimeStruct(date.getTime());
+                    if (date instanceof Date) {
+                         LocalDate dateObj = ((Date) date).toLocalDate();
+                    BMap<BString, Object> dateMap = ValueCreator.createRecordValue(
+                            org.ballerinalang.stdlib.time.util.ModuleUtils.getModule(),
+                            org.ballerinalang.stdlib.time.util.Constants.DATE_RECORD);
+                    dateMap.put(StringUtils.fromString(org.ballerinalang.stdlib.time.util.Constants.DATE_RECORD_YEAR),
+                            dateObj.getYear());
+                    dateMap.put(StringUtils.fromString(org.ballerinalang.stdlib.time.util.Constants.DATE_RECORD_MONTH),
+                            dateObj.getMonthValue());
+                    dateMap.put(StringUtils.fromString(org.ballerinalang.stdlib.time.util.Constants.DATE_RECORD_DAY),
+                            dateObj.getDayOfMonth());
+                    return dateMap;
```

Comment:
formatting issue.

Reply:
fixed in https://github.com/ballerina-platform/module-ballerina-sql/pull/161/commits/1a363a5654c30b1c3b4efd836eb9225713c4e5a6